netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -68,6 +68,34 @@ typedef struct _net_ebpf_extension_hook_provider
         LIST_ENTRY attached_clients_list; ///< Linked list of hook NPI clients that are attached to this provider.
 } net_ebpf_extension_hook_provider_t;
 
+static void
+_ebpf_ext_acquire_push_lock(_Inout_ PEX_PUSH_LOCK lock, bool exclusive)
+{
+    KeEnterCriticalRegion();
+    if (exclusive) {
+        ExAcquirePushLockExclusive(lock);
+    } else {
+        ExAcquirePushLockShared(lock);
+    }
+}
+
+static void
+_ebpf_ext_release_push_lock(_Inout_ PEX_PUSH_LOCK lock, bool exclusive)
+{
+    if (exclusive) {
+        ExReleasePushLockExclusive(lock);
+    } else {
+        ExReleasePushLockShared(lock);
+    }
+    KeLeaveCriticalRegion();
+}
```

Comment:
since this isn't inlined, this means there's an extra `if` test in the code path, which if this ever happens in a hot path will be non-optimal.  Consider either marking as inline.

Reply:
Fixed. Changed it to a macro